### PR TITLE
Update filters to use only running & healthy/no-health-check containers

### PR DESCRIPTION
**Pull request by bohanwood**
I use health check to implement blue-green deployment for zero-downtime update.
With this PR, you can do something like this
| Old Container | New Container | Action |
| ---- | ---- | ---- |
| healthy | - | - |
| healthy | starting | Start new container with compose --scale |
| healthy | unhealthy | New container started but not enabled |
| unhealthy | healthy | Route traffic to new container |
| stopped | healthy | Stop and delete old container |

For how to switch the unhealthy/healthy flag,
I refered [Kamal](https://github.com/basecamp/kamal/blob/90ecb6a12a686ddf1767152c56ef9c555a634231/lib/kamal/configuration/role.rb#L89)
 - Mount `./cord:/tmp/cord`
 - Enable new container `touch ./cord/cord`
 - Disable old container `rm -f ./cord/cord`
 
 Of course there're many other ways to implement this.
 
 Since label cannot be changed after the container is created, I afraid that health status is the only property that we can use.

### CODE REVIEW

**Comment by invzhi** [comment]
Thanks for your contribution. It is a nice patch, looks good to me.
Just need to fix the build.

File: upstreams.go [hunk 1]
```diff
@@ -129,8 +137,8 @@ func (u *Upstreams) keepUpdated(ctx caddy.Context, cli *client.Client) {
 			select {
 			case <-messages:
 				debounced(func() {
-					containers, err := cli.ContainerList(ctx, container.ListOptions{
-						Filters: filters.NewArgs(filters.Arg("label", LabelEnable)),
+					containers, err := cli.ContainerList(ctx, types.ContainerListOptions{
```
Comment (invzhi) [inline upstreams.go]: Did you use the old version of docker? I believe `types.ContainerListOptions` is deprecated.

File: upstreams.go [hunk 2]
```diff
@@ -172,8 +180,8 @@ func (u *Upstreams) Provision(ctx caddy.Context) error {
 
 	u.logger.Info("docker engine is connected", zap.String("api_version", ping.APIVersion))
 
-	containers, err := cli.ContainerList(ctx, container.ListOptions{
-		Filters: filters.NewArgs(filters.Arg("label", LabelEnable)),
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{
```
Comment (invzhi) [inline upstreams.go]: Some as before.

**Comment by bohanwood** [comment]
Fixed. The branch is based on an old version of code.

**Pull request merged**